cluster/pre-existing/util.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
+# Copyright 2017 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# A library of helper functions for landing kubemark containers on a
+# pre-existing Kubernetes master. See test/kubemark/pre-existing/README.md
+# for me details on using a pre-existing provider.
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE}")/../..
+
+source "${KUBE_ROOT}/cluster/common.sh"
+source "${KUBE_ROOT}/hack/lib/util.sh"
+
+function detect-project() {
```

Comment:
It seems to do just the job of detecting the project name (https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/util.sh#L125). All this other logic should ideally be in a separate function.
Also there is a different function named `detect-master` which does what you said (https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/util.sh#L346)